modules/@apostrophecms/user/index.js

```diff
@@ -292,6 +292,9 @@ module.exports = {
 
         await self.hashPassword(doc, safeUser);
         await self.hashSecrets(doc, safeUser);
+
+        await self.emit('beforeSaveSafe', req, safeUser, doc);
```

Comment:
I argue with myself about this order of arguments but I think you are right because `safeUser` is the thing being saved, and the doc is just additional information. 👍

Reply:
Yes at the beginning I had `user, safeUser` then I thought "wait, the document in being saved is safeUser, so user is only additional data".